accepted/00101-native-appstreams.md

```diff
@@ -0,0 +1,75 @@
+- Feature Name: Native AppStream support
+- Start Date: 2024-02-19
+
+# Summary
+[summary]: #summary
+
+This RFC proposes native support for AppStreams repositories in Uyuni.
+
+# Motivation
+[motivation]: #motivation
+
+Following the integration of [modularity and modular repositories](https://docs.fedoraproject.org/en-US/modularity/) in RHEL and its derivatives, Uyuni initially implemented modularity through content lifecycle management and the [introduction of AppStream filters](https://github.com/uyuni-project/uyuni-rfc/blob/master/accepted/00064-modular-repos-with-clm.md). These filters effectively remove modularity features from a repository, enabling consumption through the Uyuni UI. However, this approach introduced complexity and limited functionality, prompting the need for a more comprehensive solution.
+
+The goal is to seamlessly support modularity in its native form within uyuni, ensuring a user-friendly and intuitive experience across all workflows involving modular repositories.
+
+# Detailed design
+[design]: #detailed-design
+
+## Overview
+
+The design encompasses core functionality, new UI operations, modifications to the existing CLM feature, and testing infrastructure.
+
+## The core design
+
+The core design involves three key components for data collection and filtering of packages for client consuption:
+
+1. **Repo-sync**: A standalone Python module that extracts module metadata from source channels using `dnf`'s `libmodulemd` library during repo-sync, serving as the source of truth for the module information. As the data gathered during this process is critical to the feature, it must be executed as a single unit with the repo-sync ([See the PoC](https://github.com/cbbayburt/uyuni/commit/ed9391e8c6e0a66d1dd7cb0f3501332b0884f2f3)).
+
+2. **Package profile update**: A Salt state that retrieves current module information from clients, storing the data in the database. The state calls `dnf module` commands to retrieve the list of enabled module streams together with their *name*, *stream*, *version*, *context*, and *arch* values ([see the PoC](https://github.com/cbbayburt/uyuni/commit/2c788f3144f5bfe8ddd904045e0a757a7a432923)).
+
+3. **Package cache**: Incorporates data from repo-sync and package profile update into existing queries, ensuring proper filtering of non-enabled modular packages. The existing tables, queries and stored procedures will be updated with additional joins and proper indexing to ensure minimum performance impact.
+
```

Comment:
There is a 4th point needed. You need to generate new metadata for the Channels including new compiled modules.yaml from the database informations.